### DISCUSSION TITLE
Can someone please help me to write a complex query? I need to connect multiple things using an explicit many-to-many table, but only if they aren't already connected.

### DISCUSSION THREAD

**Discussion by lumenwrites** [prisma/prisma#16646] [Q&A]
Hi there! I'm looking for some help with writing a complicated query.

- I'm working on a platform for selling courses.
- A user can buy multiple courses at once.
- When the user has bought multiple courses, I want to connect all of them to the user using a single query.
- It's possible that the user already owns some of these courses. If I'm trying to connect courses that are already connected, I don't want Prisma to throw an error. I just need to connect the courses that aren't already connected.

Here's the simplified version of my database schema:
![connect](https://user-images.githubusercontent.com/4761235/205807225-6174d8a4-ba6e-4fca-8973-924749e31f6a.png)
```
model Profile {
  id                    String           @id @default(cuid())
  email                 String           @unique
  // Courses this profile owns
  courses               ProfileCourse[]
}

model Course {
  id         String          @id @default(cuid())
  slug       String          @unique @default("")
  // Profiles that own the course
  profiles   ProfileCourse[]
}

// Join table connecting Profiles and Courses
// (represents the fact that Profile has purchased a Course)
model ProfileCourse {
  createdAt           DateTime @default(now())
  profile             Profile  @relation(fields: [profileId], references: [id], onDelete: Cascade)
  profileId           String
  course              Course   @relation(fields: [courseId], references: [id])
  courseId            String

  @@id([profileId, courseId])
}
```

I'm relatively new to Prisma, and connecting things through an explicit many-to-many table is confusing, can someone please help me out?

**Comment by nurul3101** [comment]
Hello @lumenwrites 👋 
Would a query like this work for your case?
For every course that a user purchases you can check if the course with that profile exists, if it exists we do nothing, if it doesn't then we create a new record with that profileId and courseId.

We can use [upsert](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upsert) for this case with empty update as we don't want to perform any operation if an existing record is found.
```typescript
  const AddCourses = await prisma.profileCourse.upsert({
    where: {
      profileId_courseId: {
        profileId: '1',
        courseId: '1',
      },
    },
    update: {},
    create: {
      courseId: '1',
      profileId: '1',
    },
  });
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
